platforms/common/uORB/uORBManager.hpp

```diff
@@ -668,7 +668,14 @@ class Manager
 			}
 			void free() { px4_sem_destroy(&_sem); }
 			int take() { return px4_sem_wait(&_sem); }
-			int take_timedwait(struct timespec *abstime) { return px4_sem_timedwait(&_sem, abstime); }
+			int take_timedwait(struct timespec *abstime)
+			{
+#if defined(CONFIG_ARCH_BOARD_PX4_SITL)
+				return px4_sem_timedwait(&_sem, abstime);
+#else
+				return sem_clockwait(&_sem, CLOCK_MONOTONIC, abstime);
```

Comment:
px4_sem_clockwait?

Reply:
yes, could introduce that. might reduce confusion...